error-handling/src/main/java/com/palantir/remoting2/errors/ServiceException.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.errors;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeLoggable;
+import com.palantir.remoting2.ext.jackson.ObjectMappers;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.ws.rs.core.Response;
+
+public class ServiceException extends RuntimeException implements SafeLoggable {
+
+    private static final int DEFAULT_STATUS = Response.Status.INTERNAL_SERVER_ERROR.getStatusCode();
+
+    private static final ObjectMapper MAPPER = ObjectMappers.newServerObjectMapper();
+
+    private final String errorId = UUID.randomUUID().toString();
+    private final String message;
+    private final List<Arg<?>> args;
+    private final int status;
+
+    public ServiceException(String messageFormat, Arg<?>... messageParams) {
+        this(DEFAULT_STATUS, messageFormat,  messageParams);
+    }
+
+    public ServiceException(Throwable cause, String messageFormat, Arg<?>... messageParams) {
+        this(DEFAULT_STATUS, cause, messageFormat,  messageParams);
+    }
+
+    public ServiceException(int status, String messageFormat, Arg<?>... messageParams) {
+        this(status, null, messageFormat,  messageParams);
+    }
+
+    public ServiceException(int status, @Nullable Throwable cause, String message,
+            Arg<?>... args) {
+        super(formatMessage(message, args), cause);
```

Comment:
we have tests that `Exception#getMessage` does the right thing (i.e., that `formatMessage` is correct) - is that what you mean here?